### PR TITLE
Add logs for request processing to ChromeApiProvider

### PR DESCRIPTION
**Pull request by ViktoriiaKovalova**
Adds logs for requests from chrome.smartCardProviderPrivate API. Requests' arguments and results will be logged in Debug mode.

### CODE REVIEW

File: smart_card_connector_app/src/chrome-api-provider.js [hunk 1]
```diff
@@ -355,20 +358,48 @@ function convertConnectionStateToEnum(pcscState) {
...
+/**
+ * Generates a debug textual representation of the array of values.
+ *
+ * This function is safe to be used in Release builds, because all potentially
+ * privacy-sensitive data is stripped away from the resulting text.
+ * @param {!Array} arr
+ * @returns {string}
+ */
+function getArrayDebugRepresentation(arr) {
+  return goog.iter.join(goog.iter.map(arr, GSC.DebugDump.debugDump), ', ');
```
[the remark this answers is not in the document]
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js]: thanks!

File: smart_card_connector_app/src/chrome-api-provider.js [hunk 2]
```diff
@@ -355,19 +358,56 @@ function convertConnectionStateToEnum(pcscState) {
...
-  constructor(chromeEvent, callback) {
+  constructor(chromeEvent, callback, reportResultFunction, nameForLogs) {
     super();
     this.chromeEvent_ = chromeEvent;
-    this.callback_ = callback;
+    this.callback_ = async (requestId, ...args) => {
+      goog.log.fine(
+          logger,
+          `Processing an event with id ${requestId}: ${nameForLogs}(${
+              getArrayDebugRepresentation(args)})`);
```
Comment (emaxx-google) [inline smart_card_connector_app/src/chrome-api-provider.js:400]: nit: `getArrayDebugRepresentation(args)` is called twice within the same function, so can be cached in a variable instead.

Also consider doing the same for `getArrayDebugRepresentation(result)`, for readability purposes.

File: smart_card_connector_app/src/chrome-api-provider.js [hunk 3]
```diff
@@ -511,20 +575,19 @@ GSC.ConnectorApp.ChromeApiProvider = class extends goog.Disposable {
       goog.log.log(
```
Comment (emaxx-google) [inline smart_card_connector_app/src/chrome-api-provider.js:575]: These `try`-`catch` blocks seems like something that can be automatically done by some kind of a wrapper as well? Just a thought - I haven't thought it through, and no need to try it in the current PR.

**Review by emaxx-google — APPROVED** [review]
LGTM % nits

File: smart_card_connector_app/src/chrome-api-provider.js (hunk 3 above)
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js:575]: i wanted to do smth like that, but i'm not sure how can that be done nicely.
The problem is that if the error is thrown, the wrapper will need to know what are the default values it needs to return.

File: smart_card_connector_app/src/chrome-api-provider.js (hunk 2 above)
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js]: done

File: smart_card_connector_app/src/chrome-api-provider.js [hunk 4]
```diff
@@ -483,10 +548,9 @@ GSC.ConnectorApp.ChromeApiProvider = class extends goog.Disposable {
   }
 
   /**
-   * @param {number} requestId
    * @private
    */
```
[the remark this answers is not in the document]
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js:553]: done

[the remark this answers is not in the document]
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js:553]: oh wait i'm supposed to return promise from this 😮 forgot about this

[the remark this answers is not in the document]
Comment (ViktoriiaKovalova) [reply smart_card_connector_app/src/chrome-api-provider.js:553]: fixed

**Review by emaxx-google — APPROVED** [review]
LGTM

File: smart_card_connector_app/src/chrome-api-provider.js (hunk 3 above)
Comment (emaxx-google) [reply smart_card_connector_app/src/chrome-api-provider.js:576]: You can pass the default values as another parameter for that helper (`ChromeEventListener` or another helper method).

Ultimately, looking at the repetitive patterns across these methods, it seems that they can be replaced with a single generic method? It just needs to know the things that differ, something like: (1) Chrome API event name, (2) lambda that "massages" input arguments for sending them as PC/SC calls, (3) same for output arguments, (4) report*Result function, (5) default arguments for the report function.

**Pull request merged**